pubspec.yaml

```diff
@@ -7,7 +7,7 @@ environment:
   sdk: ">=2.12.0 <4.0.0"
 
 dependencies:
-  sensors_plus: ^3.0.3
+  sensors_plus: ^4.0.0
```

Comment:
4.0.0 was retracted, upgrade to 4.0.1:
https://pub.dev/packages/sensors_plus/changelog

Reply:
The caret ^ symbol means the software is compatible with all minor and patch releases that do not change the leftmost non-zero digit. So, ^4.0.0 would be compatible with any 4.x.x version

Reply:
I get that increasing the compatibility range on a package is desirable, but v4.0.0 might still be broken in some circumstances so rather I'd avoid it even in a package

Reply:
I would definitely recommend always accepting minor and patch releases in any opensource dependency.

Reply:
However, if you're really afraid that `sensors_plus` will release breaking changes in minor/patch versions (which I personally find unlikely); I would recommend at least specifying: `4.0.*` as the constraint

Reply:
I definitely agree with the ^, I'm just saying ^4.0.1 is probably better than ^4.0.0, since otherwise 4.0.0 might end up in the dependency resolution of the final project and since that version apparently has build issues in some cases (which was the reason for it being redacted).

I just skimmed through the issue linked in the redaction notice though, so perhaps it's not a big deal after all. In the end it's all just a matter of taste probably.

Reply:
Åhh, I see what you mean now